### PR TITLE
Puts the kinetic crusher's projectile firing on rmb

### PR DESCRIPTION
**Pull request by ArcaneDefence**
## About The Pull Request
Splits the crusher's afterattack so that left click is the melee hit/detonation effect and right click is to fire the projectile that marks/mines while retaining the previous behavior

## Why It's Good For The Game
It's counterintuitive that in order to hit something adjacent to you with the crusher's projectile you are required to move your mouse beyond your target.
Putting the projectile on rmb allows you to stand next to a mob or tile you want to mine and just right click on it to shoot it.

You still have to do two clicks to mark and then detonate, and this does not allow you to trigger them more rapidly.

## Changelog
:cl:
qol: Kinetic crushers now fire their projectile with the right mouse button, allowing you to fire at targets that are adjacent to you without moving your mouse away from your target.
/:cl:

### CODE REVIEW

File: code/modules/mining/equipment/kinetic_crusher.dm [hunk 1]
```diff
@@ -57,7 +57,7 @@
 
 /obj/item/kinetic_crusher/examine(mob/living/user)
 	. = ..()
-	. += span_notice("Mark a large creature with the destabilizing force, then hit them in melee to do <b>[force + detonation_damage]</b> damage.")
+	. += span_notice("Mark a large creature with the destabilizing force with right click, then hit them in melee to do <b>[force + detonation_damage]</b> damage.")
```
Comment (Fikou) [inline code/modules/mining/equipment/kinetic_crusher.dm:60]:
```suggestion
	. += span_notice("Mark a large creature with a destabilizing force with right click, then hit them in melee to do <b>[force + detonation_damage]</b> damage.")
```

File: code/modules/mining/equipment/kinetic_crusher.dm [hunk 2]
```diff
@@ -146,6 +125,41 @@
...
 
+/obj/item/kinetic_crusher/attack_secondary(atom/target, mob/living/user) //This is for aiming at yourself/adjacent mobs
+	if(target != user && wielded)
+		fire_kinetic_blast(target, user)
+	if(target == user)
+		to_chat(user, span_warning("You can't aim [src] at yourself!"))
+	if(!wielded)
+		to_chat(user, span_warning("[src] is too heavy to use with one hand! You fumble and drop everything."))
+		user.drop_all_held_items()
+	return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
```
Comment (Fikou) [inline code/modules/mining/equipment/kinetic_crusher.dm:136]: the order in this proc is nonsensical

File: code/modules/mining/equipment/kinetic_crusher.dm [hunk 3]
```diff
@@ -146,6 +125,41 @@
 					C.total_damage += detonation_damage
 				L.apply_damage(detonation_damage, BRUTE, blocked = def_check)
 
+/obj/item/kinetic_crusher/attack_secondary(atom/target, mob/living/user) //This is for aiming at yourself/adjacent mobs
```
Comment (Fikou) [inline code/modules/mining/equipment/kinetic_crusher.dm:128]: afterattack secondary works for melee range too cant see why this needs to exist

File: code/modules/mining/equipment/kinetic_crusher.dm [hunk 4]
```diff
@@ -146,6 +125,41 @@
...
+
+/obj/item/kinetic_crusher/afterattack_secondary(atom/target, mob/living/user, clickparams)
+	if(wielded)
+		fire_kinetic_blast(target, user, clickparams)
+		user.changeNext_move(CLICK_CD_MELEE)
+	return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+
+/obj/item/kinetic_crusher/proc/fire_kinetic_blast(atom/target, mob/living/user, clickparams)
+	var/modifiers = params2list(clickparams)
+	if(charged)
```
Comment (Fikou) [inline code/modules/mining/equipment/kinetic_crusher.dm:146]: early return this

**Comment by ATH1909** [comment]
god yes

File: code/modules/mining/equipment/kinetic_crusher.dm (hunk 2 above)
Comment (Fikou) [reply code/modules/mining/equipment/kinetic_crusher.dm:136]: to elaborate:
you shouldnt be checking
if case 1 and case 2 succeed
if case 1 fail
if case 2 fail
this is inefficient, youre doing checks twice and it looks dumb
way better is
if case 1 fail
if case 2 fail
succeed

File: code/modules/mining/equipment/kinetic_crusher.dm [hunk 5]
```diff
@@ -146,6 +125,39 @@
...
+	if(!charged)
+		return
+	var/modifiers = params2list(clickparams)
+	var/turf/proj_turf = user.loc
+	if(!isturf(proj_turf))
+		return
+	var/obj/projectile/destabilizer/D = new /obj/projectile/destabilizer(proj_turf)
+	for(var/t in trophies)
+		var/obj/item/crusher_trophy/T = t
+		T.on_projectile_fire(D, user)
```
Comment (Fikou) [inline code/modules/mining/equipment/kinetic_crusher.dm:151]:
```suggestion
	var/obj/projectile/destabilizer/destabilizer = new(proj_turf)
	for(var/obj/item/crusher_trophy/trophy as anything in trophies)
		trophy.on_projectile_fire(destabilizer, user)
```

Comment (Fikou) [reply code/modules/mining/equipment/kinetic_crusher.dm:151]: and use destabilizer for the other parts of the proc too

File: code/modules/mining/equipment/kinetic_crusher.dm [hunk 6]
```diff
@@ -146,6 +125,39 @@
...
 				L.apply_damage(detonation_damage, BRUTE, blocked = def_check)
 
+/obj/item/kinetic_crusher/attack_secondary(atom/target, mob/living/user, clickparams)
+	return SECONDARY_ATTACK_CONTINUE_CHAIN
+
+/obj/item/kinetic_crusher/afterattack_secondary(atom/target, mob/living/user, clickparams)
+	if(!wielded)
+		return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+	if(target == user)
+		to_chat(user, span_warning("You can't aim [src] at yourself!"))
```
Comment (Fikou) [inline code/modules/mining/equipment/kinetic_crusher.dm:135]: balloon alert this maybe?

File: code/modules/mining/equipment/kinetic_crusher.dm [hunk 7]
```diff
@@ -146,6 +125,39 @@
 					C.total_damage += detonation_damage
 				L.apply_damage(detonation_damage, BRUTE, blocked = def_check)
 
+/obj/item/kinetic_crusher/attack_secondary(atom/target, mob/living/user, clickparams)
+	return SECONDARY_ATTACK_CONTINUE_CHAIN
+
+/obj/item/kinetic_crusher/afterattack_secondary(atom/target, mob/living/user, clickparams)
+	if(!wielded)
+		return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
```
Comment (Fikou) [inline code/modules/mining/equipment/kinetic_crusher.dm:133]: balloon alert or a chat message here?

**Pull request merged**